PatientWebApplication/AppointmentMicroserviceApi/Service/RegularAppointmentService.cs

```diff
@@ -0,0 +1,390 @@
+        /// <returns> list of all available appointments on specific date for given doctor and patient. </returns>
+        public async Task<List<DoctorAppointment>> GetAllAvailableAppointmentsForDateAsync(string dateString, int doctorId, int patientId)
+        {
+            MicroserviceShiftDto doctorShift = await Utility.HttpRequests.GetShiftForDoctorForSpecificDay(new DoctorShiftSearchDto(doctorId, dateString));
+            if (doctorShift == null)
+            {
+                return new List<DoctorAppointment>();
+            }
+            return CreateListOfAvailableAppointments(doctorShift, dateString, doctorId, patientId);
+        }
+
+        private List<DoctorAppointment> CreateListOfAvailableAppointments(MicroserviceShiftDto doctorShift, string dateString, int doctorId, int patientId)
+        {
+            List<DoctorAppointment> availableAppointments = new List<DoctorAppointment>();
+            TimeSpan time = TimeSpan.Parse(doctorShift.StartTime);
+            while (time != TimeSpan.Parse(doctorShift.EndTime))
+            {
+                if (time < DateTime.Now.TimeOfDay && UtilityMethods.CheckIfDateIsToday(dateString))
+                {
+                    time = time.Add(TimeSpan.FromMinutes(15));
+                    continue;
+                }
+                AddAppointments(time, availableAppointments, dateString, patientId, doctorId);
+                time = time.Add(TimeSpan.FromMinutes(15));
+            }
+            return availableAppointments;
+        }
+
+        private void AddAppointments(TimeSpan time, List<DoctorAppointment> availableAppointments, string dateString, int patientId, int doctorId)
+        {
+            DateTime date = UtilityMethods.ParseDateInCorrectFormat(dateString);
+            List<TimeSpan> startTimesAppointments = GetAllStartTimes(CreateAppointmentSetForDate(date, doctorId, patientId).ToList());
+            List<Operation> operations = operationService.CreateOperationtSetForDate(date, doctorId, patientId).ToList();
+            if (!startTimesAppointments.Contains(time) && !operationService.IsOperationInTimePeriod(time, operations))
+            {
+                availableAppointments.Add(new DoctorAppointment(0, time, dateString, patientId, doctorId, new List<Referral>(), Utility.HttpRequests.GetDoctorByIdAsync(doctorId).Result.Ordination));
+            }
+        }
+
+        /// <summary> This method is creating a set out of list of doctor's and patient's apaintments on specific date. </summary>
+        /// <returns> <c>HashSet</c> of appointments. </returns>
+        private HashSet<DoctorAppointment> CreateAppointmentSetForDate(DateTime date, int doctorId, int patientId)
+        {
+            HashSet<DoctorAppointment> appointmentsSet = new HashSet<DoctorAppointment>(GetAllAppointmentsByDateAndDoctor(date, doctorId));
+            appointmentsSet.UnionWith(GetAllAppointmentsByDateAndPatient(date, patientId));
+            return appointmentsSet;
```

Comment:
Write this in form:
"
return appointmentsSet.UnionWith(GetAllAppointmentsByDateAndPatient(date, patientId));
"